frontend/__tests__/inquiry/add_inquiry.test.tsx

```diff
@@ -0,0 +1,94 @@
+/* Codacy is unhappy with await */
+/* eslint-disable @typescript-eslint/await-thenable */
+/* Codacy is unhappy with click events */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* Codacy is unhappy with mocking */
+/* eslint-disable @typescript-eslint/no-empty-function */
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+
+import { fireEvent, render, screen } from "@testing-library/react"
+import userEvent from "@testing-library/user-event"
+import { Inquiries } from "../../src/routes/_layout/inquiries"
+import "@testing-library/jest-dom"
+import {
+  MAX_INQUIRY_LENGTH,
+  MIN_INQUIRY_LENGTH,
+} from "../../src/components/Inquiries/AddInquiry"
+
+const unicodeText =
+  "Тенденция к взаимопомощи у человека имеет столь отдаленное происхождение и так глубоко переплетена со всей прошлой эволюцией человеческого рода, что она сохранилась у человечества вплоть до настоящего времени, несмотря на все превратности истории."
+const nonUnicodeText =
+  "\uD800-\uDBFF\uD800-\uDBFF\uD800-\uDBFF\uD800-\uDBFF\uD800-\uDBFF\uD800-\uDBFF\uD800-\uDBFF\uD800-\uDBFF\uD800-\uDBFF"
+
+jest.mock("../../src/components/Inquiries/InquiriesTable", () => ({
+  __esModule: true,
+  default: () => <div />,
+}))
+
+jest.mock("@tanstack/react-query", () => ({
+  ...jest.requireActual("@tanstack/react-query"),
+  useQueryClient: () => {},
+  useMutation: () => ({
+    mutate: () => {},
+  }),
+}))
+
+describe("Add Inquiry", () => {
+  beforeEach(async () => {
+    render(<Inquiries />)
+    await userEvent.click(screen.getByText("Add Inquiry"))
+  })
+
+  it("should display add modal when user presses Add Inquiry button", async () => {
+    const textArea = await screen.getByTestId("add-inquiry-text")
+    fireEvent.change(textArea, {
+      target: {
+        value: "Why do birds suddenly appear every time you are near?",
+      },
+    })
+    await userEvent.click(screen.getByTestId("submit-add-inquiry"))
```

Comment:
confused -- so how does it verify that the modal appeared?

Reply:
it waits for the modal to allow entry of the Inquiry.  You're actually right that I am not testing the Toast that tells you whether the Inquiry was submitted successfully to the backend.  For a few reasons:  That is best handled by an integration or e2e test.  The confirmation messages after Inquiry submission or failure are built into the boilerplate we are using.  We are making use of their components rather than our code so I didn't see it as being essential for us to test it


```typescript

const AddInquiry = ({isOpen, onClose}: AddInquiryProps) => {
    const queryClient = useQueryClient()
    const showToast = useCustomToast()

    // already typed by react-hook-form https://react-hook-form.com/docs/useform#errors
    // eslint-disable-next-line
    const {
        register,
        handleSubmit,
        reset,
        formState: {errors, isSubmitting},
        /* eslint-disable-next-line  */
    } = useForm<InquiryCreate>({
        mode: "onBlur",
        criteriaMode: "all",
        defaultValues: {
            text: "",
        },
    })

    const mutation = useMutation({
        mutationFn: (data: InquiryCreate) =>
            InquiriesService.createInquiry({requestBody: data}),
        onSuccess: () => {
            showToast("Success!", "Inquiry created successfully.", "success")
            /* eslint-disable-next-line  */
            reset()
            onClose()
        },
        onError: (err: ApiError) => {
            handleError(err, showToast)
        },
        onSettled: async () => {
            await queryClient.invalidateQueries({queryKey: ["inquiries"]});
        },
    })

    const onSubmit: SubmitHandler<InquiryCreate> = (data) => {
        /* eslint-disable-next-line  */
        mutation.mutate(data)
    }
```